src/tempstore-posix.c

```diff
@@ -0,0 +1,309 @@
+
+    qdts_file_t *file = DEQ_HEAD(state->files);
+    while (!!file && strcmp(file->file_path, full_path) != 0) {
+        file = DEQ_NEXT(file);
+    }
+
+    if (!file && DEQ_SIZE(state->files) >= limit && limit != 0) {
+        free(full_path);
+        return false;
+    }
+
+    int fd = open(full_path, O_CREAT | O_TRUNC | O_WRONLY, S_IRUSR | S_IWUSR);
+    if (fd > 0) {
+        ssize_t written = write(fd, body, strlen(body));
+        if (written < 0) {
+            qd_log(state->log, QD_LOG_ERROR, "Failed to write file '%s' (%s)", full_path, strerror(errno));
+            free(full_path);
+            return true;
+        }
+        close(fd);
+    } else {
+        qd_log(state->log, QD_LOG_ERROR, "Failed to open file '%s' (%s)", full_path, strerror(errno));
+        free(full_path);
+        return true;
+    }
+
+    if (!!file) {
+        file->write_count++;
+        free(full_path);
+    } else {
+        file = NEW(qdts_file_t);
+        DEQ_ITEM_INIT(file);
+        file->file_path   = full_path;
+        file->write_count = 1;
+        DEQ_INSERT_TAIL(state->files, file);
+    }
+
+    qd_log(state->log, QD_LOG_INFO, "Temporary file stored: %s", filename);
+    return true;
+}
+
+
+static uint64_t qdts_on_message(void *context, qd_message_t *msg, int link_maskbit, int inter_router_cost,
+                                uint64_t conn_id, const qd_policy_spec_t *policy_spec, qdr_error_t **error)
+{
+    qdts_state_t *state = (qdts_state_t*) context;
+
+    if (!!policy_spec && !policy_spec->allowTempFile) {
+        qd_log(state->log, QD_LOG_ERROR, "[C%"PRIu64"] Policy violation - temp file store not permitted", conn_id);
+        *error = qdr_error(QD_AMQP_COND_UNAUTHORIZED_ACCESS, "File store access forbidden");
+        return PN_REJECTED;
+    }
+
+    if (state->failed) {
+        *error = qdr_error(QD_AMQP_COND_INTERNAL_ERROR, "File store initialization failed");
+        return PN_REJECTED;
+    }
+
+    if (qd_message_check_depth(msg, QD_DEPTH_BODY) != QD_MESSAGE_DEPTH_OK) {
+        qd_log(state->log, QD_LOG_ERROR, "Invalid message received");
+        *error = qdr_error(QD_AMQP_COND_DECODE_ERROR, "Parse error in message");
+        return PN_REJECTED;
+    }
+
+    qd_iterator_t *subject_iter = qd_message_field_iterator(msg, QD_FIELD_SUBJECT);
+    if (!subject_iter) {
+        qd_log(state->log, QD_LOG_ERROR, "Invalid message received - No Subject");
+        *error = qdr_error(QD_AMQP_COND_INVALID_FIELD, "Invalid request - no subject header");
+        return PN_REJECTED;
+    }
+
+    char *filename = (char*) qd_iterator_copy(subject_iter);
+    qd_iterator_free(subject_iter);
+
+    if (!!strchr(filename, '/') || !!strchr(filename, '\\') || !!strstr(filename, "..")) {
+        qd_log(state->log, QD_LOG_ERROR, "Invalid message received - File name with path traversal characters");
+        free(filename);
+        *error = qdr_error(QD_AMQP_COND_INVALID_FIELD, "File name contains path traversal characters");
```

Comment:
Can we please mention in the qdrouter.json that file name containing path traversal characters are invalid?